zerver/tests/test_handle_push_notification.py

```diff
@@ -148,7 +148,8 @@ def test_end_to_end(self) -> None:
 
     @activate_push_notification_service()
     @responses.activate
-    def test_end_to_end_failure_due_to_no_plan(self) -> None:
+    @mock.patch("zerver.lib.push_notifications.send_push_notifications")
+    def test_end_to_end_failure_due_to_no_plan(self, unused_mock: mock.MagicMock) -> None:
```

Comment:
When implementing settings to make E2EE mandatory, we can write a test where "E2EE is not required" and "both type of devices exist" - registered with legacy system and the newer one.

Reply:
Sounds reasonable to me.